core/src/storage2/collections/binary_heap/elements.rs

```diff
@@ -0,0 +1,291 @@
+    /// Appends an element to the back of the vector.
+    pub fn push(&mut self, value: T) {
+        assert!(
+            self.len() < core::u32::MAX,
+            "cannot push more elements into the storage vector"
+        );
+        let last_index = self.len();
+        *self.len += 1;
+        self.push_to(last_index, Some(value));
+    }
+
+    fn push_to(&mut self, index: u32, value: Option<T>) {
+        let children_index = children::get_children_storage_index(index);
+        match self.children.get_mut(children_index) {
+            Some(children) => {
+                let child_pos = children::get_child_pos(index);
+                *children.child_mut(child_pos) = value;
+            }
+            None => {
+                self.children.push(Children::new(value, None));
+                debug_assert!(
+                    self.children.get(children_index).is_some(),
+                    "the new children were not placed at children_index!"
+                );
+            }
+        };
+    }
+
+    /// Pops the last element from the vector and returns it.
+    //
+    /// Returns `None` if the vector is empty.
+    pub fn pop(&mut self) -> Option<T> {
+        if self.is_empty() {
+            return None
+        }
+        let last_index = self.len() - 1;
+        *self.len = last_index;
+
+        let children_index = children::get_children_storage_index(last_index);
+        let old = self.children.get_mut(children_index);
+        match old {
+            Some(children) => {
+                let child_pos = children::get_child_pos(last_index);
+                let popped_val = children.child_mut(child_pos).take();
+                if children.is_empty() {
+                    // if both children are non-existent the entire children object can be removed
+                    self.children.pop();
+                }
+                popped_val
+            }
+            None => {
+                unreachable!("vec must contain children at children_index of last_index")
+            }
+        }
+    }
+
+    /// Returns the index if it is within bounds or `None` otherwise.
+    fn within_bounds(&self, index: u32) -> Option<u32> {
+        if index < self.len() {
+            return Some(index)
+        }
+        None
+    }
+}
+
+impl<T> SpreadLayout for Elements<T>
+where
+    T: SpreadLayout + Ord + PackedLayout,
+{
+    const FOOTPRINT: u64 = 1 + <StorageVec<Children<T>> as SpreadLayout>::FOOTPRINT;
+
+    fn pull_spread(ptr: &mut KeyPtr) -> Self {
+        let len = SpreadLayout::pull_spread(ptr);
+        let children = SpreadLayout::pull_spread(ptr);
+        Self { len, children }
+    }
+
+    fn push_spread(&self, ptr: &mut KeyPtr) {
+        SpreadLayout::push_spread(&self.len, ptr);
+        SpreadLayout::push_spread(&self.children, ptr);
+    }
+
+    fn clear_spread(&self, ptr: &mut KeyPtr) {
+        SpreadLayout::clear_spread(&self.len, ptr);
+        SpreadLayout::clear_spread(&self.children, ptr);
+    }
+}
```

Comment:
Please remove all methods from `Elements` that are not definitely needed by the `BinaryHeap` implementation in `mod.rs`. At least to me it seems like there are some unnecessary left overs from the copy of storage `Vec`.